awx/ui_next/src/screens/Template/shared/WorkflowJobTemplateForm.jsx

```diff
@@ -0,0 +1,473 @@
+import CheckboxField from '../../../components/FormField/CheckboxField';
+
+function WorkflowJobTemplateForm({
+  handleSubmit,
+  handleCancel,
+  i18n,
+  template = {},
+  webhook_key,
+  submitError,
+}) {
+  const { id } = useParams();
+  const wfjtAddMatch = useRouteMatch('/templates/workflow_job_template/add');
+
+  const urlOrigin = window.location.origin;
+
+  const [contentError, setContentError] = useState(null);
+  const [webhookKey, setWebHookKey] = useState(webhook_key);
+  const [credTypeId, setCredentialTypeId] = useState();
+  const [webhookService, setWebHookService] = useState(
+    template.webhook_service || ''
+  );
+  const [hasWebhooks, setHasWebhooks] = useState(Boolean(webhookService));
+
+  const webhookServiceOptions = [
+    {
+      value: '',
+      key: '',
+      label: i18n._(t`Choose a Webhook Service`),
+      isDisabled: true,
+    },
+    {
+      value: 'github',
+      key: 'github',
+      label: i18n._(t`GitHub`),
+      isDisabled: false,
+    },
+    {
+      value: 'gitlab',
+      key: 'gitlab',
+      label: i18n._(t`Git Lab`),
+      isDisabled: false,
+    },
+  ];
+
+  useEffect(() => {
+    if (!webhookService) {
+      return;
+    }
+    const loadCredentialType = async () => {
+      try {
+        const {
+          data: { results },
+        } = await CredentialTypesAPI.read({
+          namespace: `${webhookService}_token`,
+        });
+        setCredentialTypeId(results[0].id);
+      } catch (err) {
+        setContentError(err);
+      }
+    };
+    loadCredentialType();
+  }, [webhookService]);
+
+  const changeWebhookKey = async () => {
+    try {
+      const {
+        data: { webhook_key: key },
+      } = await WorkflowJobTemplatesAPI.updateWebhookKey(id);
+      setWebHookKey(key);
+    } catch (err) {
+      setContentError(err);
+    }
+  };
+
+  let initialWebhookKey = webhook_key;
+
+  const setWebhookValues = (form, webhookServiceValue) => {
+    if (
+      webhookServiceValue === form.initialValues.webhook_service ||
+      webhookServiceValue === ''
+    ) {
+      form.setFieldValue(
+        'webhook_credential',
+        form.initialValues.webhook_credential
+      );
+      form.setFieldValue('webhook_url', form.initialValues.webhook_url);
+      form.setFieldValue('webhook_service', form.initialValues.webhook_service);
+
+      setWebHookKey(initialWebhookKey);
+    } else {
+      form.setFieldValue('webhook_credential', null);
+      form.setFieldValue(
+        'webhook_url',
+        `${urlOrigin}/api/v2/workflow_job_templates/${template.id}/${webhookServiceValue}/`
+      );
+
+      setWebHookKey('A NEW WEBHOOK KEY WILL BE GENERATED ON SAVE');
+    }
+  };
+
+  const handleWebhookEnablement = (
+    form,
+    enabledWebhooks,
+    webhookServiceValue
+  ) => {
+    if (!enabledWebhooks) {
+      initialWebhookKey = webhookKey;
+      form.setFieldValue('webhook_credential', null);
+      form.setFieldValue('webhook_service', '');
```

Comment:
I think for https://github.com/ansible/awx/pull/5841/files#r383352280 I can actually reset this to the initial value and will be able to remove the || here https://github.com/ansible/awx/pull/5841/files#r383352280